src/index.ml

```diff
@@ -214,19 +226,22 @@ module Make (K : Key) (V : Value) (IO : IO) = struct
           let highest_entry = get_entry_iff_needed t i high highest_entry in
           let lowest_hash = float_of_int (K.hash lowest_entry.key) in
           let highest_hash = float_of_int (K.hash highest_entry.key) in
-          if lowest_hash > hashed_key || highest_hash < hashed_key then None
+          if lowest_hash > hashed_keyf || highest_hash < hashed_keyf then None
```

Comment:
This comparison should work with ints. Float values are only used in computing `doff`, so you could move the conversions there.